File: Assets/PlayerController.js
#pragma strict
public var speed : float = 3;
public var jumpPower : float = 6;

private var direction : Vector3 = Vector3.zero;
private var playerController : CharacterController;

function Start () {
	playerController = GetComponent(CharacterController);
	
		Debug.Log(playerController);
	
}

function Update () {
	if (playerController.isGrounded) {
		var inputX : float = Input.GetAxis( "Horizontal" );
		var inputY : float = Input.GetAxis( "Vertical" );
		
		var inputDirection : Vector3 = Vector3( inputX, 0, inputY );
		direction = Vector3.zero;
		
		if (inputDirection.magnitude > 0.1) {
			transform.LookAt( transform.position + inputDirection );
			direction += transform.forward * speed;
		}
		
		if (Input.GetButton("Jump")) {
			Debug.Log("Jump!!");
			Debug.Log(direction.y);
			
			direction.y += jumpPower;
			
			Debug.Log(direction.y);
		}
		
	}

	Debug.Log(direction);
	direction.y = Physics.gravity.y * Time.deltaTime;
	Debug.Log(direction);
	playerController.Move(direction * Time.deltaTime);
}

File: Assets/Scripts/PlayerController.cs
using UnityEngine;
using System.Collections;

public class PlayerController : MonoBehaviour 
{
	public float speed = 3;
	public float jumpPower = 6;

	private Vector3 direction = Vector3.zero;
	private CharacterController playerController;
	private Animator animetor;

	// Use this for initialization
	void Start () 
	{
		playerController = GetComponent<CharacterController>();
		animetor = GetComponentInChildren<Animator> ();
	}
	
	// Update is called once per frame
	void Update () 
	{
		if (playerController.isGrounded) {
			float inputX = Input.GetAxis ("Horizontal");
			float inputY = Input.GetAxis ("Vertical");

			Vector3 inputDirection = new Vector3 (inputX, 0, inputY);
			direction = Vector3.zero;

			if (inputDirection.magnitude > 0.1) {
				transform.LookAt (transform.position + inputDirection);
				direction += transform.forward * speed;
				animetor.SetFloat ("Speed", direction.magnitude);
			} else {
				animetor.SetFloat ("Speed", 0);
			}

			if (Input.GetButton("Jump")) {
				direction.y += jumpPower;
			}
		}

		direction.y += Physics.gravity.y * Time.deltaTime;
		playerController.Move (direction * Time.deltaTime);
	}
}
